static/js/components/WeatherWidget.jsx

```diff
@@ -0,0 +1,246 @@
+  sunset = now.diff(sunset, "hour") >= 12 ? sunset.add("1", "day") : sunset;
+  const description = weather?.weather?.current.weather[0].description || "N/A";
+
+  return (
+    <>
+      <div className={styles.weatherBar}>
+        <div className={styles.weatherInfo}>
+          {weather.weather && (
+            <>
+              <div>
+                <img src={url} className={styles.media} alt={description} />
+              </div>
+              <div className={styles.description}>
+                <Typography variant="body2" color="textSecondary" component="p">
+                  It&nbsp;is&nbsp;
+                  {(weather?.weather?.current.temp - 273.15).toFixed(1)}&deg;C
+                  with&nbsp;
+                  {weather?.weather?.current.humidity}% humidity &amp;&nbsp;
+                  {weather?.weather?.current.weather[0].description}.&nbsp;
+                  {sunrise.isBefore(sunset) && (
+                    <span>
+                      Sunrise {dayjs().to(sunrise)}, sunset {dayjs().to(sunset)}
+                      .
+                    </span>
+                  )}
+                  {!sunrise.isBefore(sunset) && (
+                    <span>
+                      Sunset {dayjs().to(sunset)}, sunrise&nbsp;
+                      {dayjs().to(sunrise)}
+                    </span>
+                  )}
+                </Typography>
+              </div>
+            </>
+          )}
+          {!weather.weather && <p>No weather information available</p>}
+        </div>
+        <CardActions className={styles.weatherLinks}>
+          {weather?.weather_link && (
+            <a href={weather.weather_link} rel="noreferrer" target="_blank">
+              <Button size="small" color="primary">
+                Forecast
+              </Button>
+            </a>
+          )}
+          {weather?.skycam_link && (
+            <a href={weather.skycam_link} rel="noreferrer" target="_blank">
+              <Button size="small" color="primary">
+                Webcam
+              </Button>
+            </a>
+          )}
+        </CardActions>
+      </div>
+    </>
+  );
+};
+
+const WeatherWidget = ({ classes }) => {
+  const styles = useStyles();
+
+  const dispatch = useDispatch();
+  const weather = useSelector((state) => state.weather.weather);
+  const userPrefs = useSelector((state) => state.profile.preferences.weather);
+  const telescopeList = useSelector((state) => state.telescopes.telescopeList);
+
+  const weatherPrefs = userPrefs || defaultPrefs;
+  const [anchorEl, setAnchorEl] = useState(null);
+
+  useEffect(() => {
+    // eslint-disable-next-line no-unused-vars
+    const getWeatherData = () => {
+      dispatch(fetchWeather(weatherPrefs.telescopeID));
+    };
+  }, [weatherPrefs, dispatch]);
+
+  const handleClose = () => {
+    setAnchorEl(null);
+  };
+
+  const handleMenuItemClick = (event, index) => {
+    const prefs = {
+      weather: { telescopeID: index },
```

Comment:
`index` was a bit confusing to me
```suggestion
  const handleMenuItemClick = (event, telescopeID) => {
    const prefs = {
      weather: { telescopeID },
```